rotkehlchen/tests/unit/test_tasks_manager.py

```diff
@@ -379,13 +390,22 @@ def test_update_snapshot_balances(rotkehlchen_instance: 'Rotkehlchen'):
             assert save_tokens_mock.call_count == 3
             assert save_tokens_mock.call_args_list[0].kwargs['address'] == accounts[1]
             assert save_tokens_mock.call_args_list[0].kwargs['blockchain'] == SupportedBlockchain.ETHEREUM  # noqa: E501
-            assert save_tokens_mock.call_args_list[0].kwargs['tokens'] == [A_DAI]
+            assert set(save_tokens_mock.call_args_list[0].kwargs['tokens']) == {A_COMP, A_LUSD, A_DAI}  # noqa: E501
             assert save_tokens_mock.call_args_list[1].kwargs['address'] == accounts[2]
             assert save_tokens_mock.call_args_list[1].kwargs['blockchain'] == SupportedBlockchain.OPTIMISM  # noqa: E501
             assert save_tokens_mock.call_args_list[1].kwargs['tokens'] == [A_USDC]
     except gevent.Timeout as e:
         raise AssertionError(f'Update snapshot balances was not completed within {timeout} seconds') from e  # noqa: E501
 
+    # verify that newly saved tokens respects the list of already existing tokens
```

Comment:
this test failed before your fix right?

Reply:
Yes, test failed before the change. The only tokens kept where the ones detected form the events and any other token was discarded. After the fix we keep the tokens already there